client/blocks/login/index.jsx

```diff
@@ -145,33 +143,23 @@ class Login extends Component {
 		}
 	};
 
-	rebootAfterLogin = () => {
-		const { redirectTo } = this.props;
-
+	rebootAfterLogin = async () => {
 		this.props.recordTracksEvent( 'calypso_login_success', {
 			two_factor_enabled: this.props.twoFactorEnabled,
 			social_service_connected: this.props.socialConnect,
 		} );
 
 		// Redirects to / if no redirect url is available
-		const url = redirectTo ? redirectTo : window.location.origin;
+		const url = this.props.redirectTo || '/';
 
 		// User data is persisted in localstorage at `lib/user/user` line 157.
 		// Only clear the data if a user is currently set, otherwise keep the
```

Comment:
Is the statement in this comment still true? In particular, does signup reuse it?

Reply:
Yes, the signup state has one special property: it is transferred from the logged out session into the logged in session. All other state slices are reset. I'm not sure if that property is used anywhere at this moment, but will certainly be useful again soon if it isn't.

This login code prevents clearing the logged out session persisted data after successful login. It only clears the data when some user A is already logged in and then user B logs in.